tests/testthat/test_preprocess_count_data.R

```diff
@@ -0,0 +1,185 @@
+  )
+})
+
+# Test data setup
+hosp_data <- tibble::tibble(
+  date = lubridate::ymd(c("2023-11-01", "2023-11-02")),
+  daily_admits = c(10, 20),
+  state_pop = c(1e6, 1e6)
+)
+
+# Test that function returns a dataframe with correct columns
+test_that("Function returns dataframe with correct columns", {
+  processed <- preprocess_count_data(hosp_data,
+    count_col_name = "daily_admits",
+    pop_size_col_name = "state_pop"
+  )
+
+  expected_cols <- c("date", "count", "total_pop")
+  expect_true(checkmate::check_names(names(processed),
+    must.include = expected_cols
+  ))
+})
+
+# Test that count column is renamed correctly
+test_that("Count column is renamed correctly", {
+  processed <- preprocess_count_data(hosp_data,
+    count_col_name = "daily_admits",
+    pop_size_col_name = "state_pop"
+  )
+
+  expect_false("daily_admits" %in% names(processed))
+  expect_true(checkmate::check_names(names(processed), must.include = "count"))
+})
+
+# Test that population size column is renamed correctly
+test_that("Population size column is renamed correctly", {
+  processed <- preprocess_count_data(hosp_data,
+    count_col_name = "daily_admits",
+    pop_size_col_name = "state_pop"
+  )
+
+  expect_false("state_pop" %in% names(processed))
+  expect_true("total_pop" %in% names(processed))
+  expect_true(checkmate::check_names(names(processed),
+    must.include = "total_pop"
+  ))
+})
+
+test_that("Function handles missing columns with an error", {
+  incomplete_hosp_data <- hosp_data |> dplyr::select(-daily_admits)
+
+
+  expect_error(preprocess_count_data(incomplete_hosp_data,
+    count_col_name = "daily_admits",
+    pop_size_col_name = "state_pop"
+  ))
+})
+
+test_that("All rows are preserved after preprocessing", {
+  processed <- preprocess_count_data(hosp_data,
+    count_col_name = "daily_admits",
+    pop_size_col_name = "state_pop"
+  )
+
+  # The number of rows should remain unchanged after processing
+  expect_equal(nrow(processed), nrow(hosp_data))
+})
+
+# Test that no NA values are introduced during preprocessing
+# (assuming input has no NAs)
+test_that("No NA values are introduced during preprocessing", {
+  processed <- preprocess_count_data(hosp_data,
+    count_col_name = "daily_admits",
+    pop_size_col_name = "state_pop"
+  )
+
+  # Check for any new NA values introduced in any column
+  expect_true(all(complete.cases(processed)))
+})
+
+# Test that renaming columns works when using default parameter values
+test_that("Renaming columns works with default parameters", {
+  test_hosp_data <- tibble::tibble(
+    date = lubridate::ymd(c("2023-11-01", "2023-11-02")),
+    daily_hosp_admits = c(10, 20),
+    state_pop = c(1e6, 1e6)
+  )
+  # Using default column names
+  default_processed <- preprocess_count_data(test_hosp_data)
+
+  expect_false(any(c(
+    "daily_admits", "state_pop"
+  ) %in% names(default_processed)))
+  expect_true(all(c("count", "total_pop") %in% names(default_processed)))
```

Comment:
```suggestion
  checkmate::expect_names(
      names(default_processed), 
      must.include = c("count", "total_pop") , 
      disjoint.from = c("daily_admits", "state_pop"))
```

Reply:
OK just confirmed that this does in fact error. The argument is called `disjunct.from = ` not `disjoint.from `

Reply:
Apologies, typing from memory!